### PR TITLE
feat: [SRM-10723]: add microfrontend components for Error Tracking

### PR DESCRIPTION
**Pull request by DFieldFL**
##### Summary:

Use Error Tracking microfrontend component the ET Event List

##### Jira Links:

https://harness.atlassian.net/browse/SRM-10723

##### Screenshots:

The full Error List in Monitored Services has been replaced with the ET microfrontend component
Screen Shot 2022-06-08 at 8 39 31 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by DFieldFL** [comment]
retrigger eslint

**Comment by DFieldFL** [comment]
retrigger coverage

**Comment by DFieldFL** [comment]
retrigger eslint
retrigger jest
retrigger typecheck
retrigger build
retrigger standards

**Comment by DFieldFL** [comment]
retrigger eslint

**Comment by DFieldFL** [comment]
retrigger build

**Comment by DFieldFL** [comment]
retrigger jest

**Comment by DFieldFL** [comment]
retrigger standards

**Comment by DFieldFL** [comment]
retrigger typecheck

File: src/modules/85-cv/ErrorTrackingApp.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  environmentId: string
+  versionId?: string
+  fromDateTime: number
+  toDateTime: number
+  routeDefinitions: Record<string, unknown>
+}
+
+export const ErrorTrackingRemoteComponentMounter: React.FC<RemoteMounterProps> = props => {
+  const { path } = useRouteMatch()
+  const { getToken: useGetToken } = SessionToken
```
Comment (abhinavrastogi-harness) [inline src/modules/85-cv/ErrorTrackingApp.tsx:45]: `getToken` is a plain function, not a hook. Renaming it to `useGetToken` is misleading, and will unnecesarily apply "Rules of Hooks", like not allowing conditional execution.

File: configs/devServerProxy.config.js [hunk 2]
```diff
@@ -112,6 +112,9 @@ module.exports = {
     pathRewrite: { '^/et': '' },
     target: process.env.ERROR_TRACKING_URL || 'http://localhost:9191'
   },
+  '/api/dashboard': {
```
Comment (abhinavrastogi-harness) [inline configs/devServerProxy.config.js:115]: we follow the pattern of `{prefix}/api`.

Comment (abhinavrastogi-harness) [inline configs/devServerProxy.config.js:115]: `dashboard` is a very generic term, and might clash with other dashboard apis we have. Can we make it more specific to CV/ET?

Comment (abhinavrastogi-harness) [inline configs/devServerProxy.config.js:115]: why is UI and API both proxying to the same URL? is the same server serving both?

File: src/global.d.ts [hunk 3]
```diff
@@ -104,6 +104,19 @@ declare module 'errortracking/App' {
   export default ChildApp
 }
 
+declare module 'errortracking/RemoteComponentWrapper' {
```
Comment (abhinavrastogi-harness) [inline src/global.d.ts:107]: we should not be requiring a wrapper to be imported back in from a MFE. 
Ideally we should be using the existing ChildAppMounter. If there's some issue with that, let's discuss.
Or move the custom mounter to NGUI.

File: src/global.d.ts [hunk 4]
```diff
@@ -104,6 +104,19 @@ declare module 'errortracking/App' {
...
+  import type { RemoteComponentWrapperProps } from '@cv/ErrorTrackingApp'
+  const ChildApp: React.ComponentType<RemoteComponentWrapperProps>
+  export default ChildApp
+}
+
+declare module 'errortracking/EventList' {
```
Comment (abhinavrastogi-harness) [inline src/global.d.ts:113]: why is `errortracking/EventList` a separate module from `errortracking/App`? Can't ET module be contained within a single module that's loaded once?

File: configs/devServerProxy.config.js (hunk 2 above)
Comment (DFieldFL) [reply configs/devServerProxy.config.js:115]: will remove

Comment (DFieldFL) [reply configs/devServerProxy.config.js:115]: will remove

Comment (DFieldFL) [reply configs/devServerProxy.config.js:115]: will remove

File: src/modules/85-cv/ErrorTrackingApp.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  const params = useParams<ProjectPathProps>()
+
+  return (
+    <RemoteComponentWrapper
```
[the remark this answers is not in the document]
Comment (DFieldFL) [reply src/modules/85-cv/ErrorTrackingApp.tsx]: using it now

File: src/global.d.ts (hunk 3 above)
Comment (DFieldFL) [reply src/global.d.ts]: using ChildAppMounter

File: src/global.d.ts (hunk 4 above)
Comment (DFieldFL) [reply src/global.d.ts]: Took the approach based on your comments

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by mdamour-harness — APPROVED** [review]

**Review by vasuki-kalluri — DISMISSED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by vasuki-kalluri — DISMISSED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by vasuki-kalluri — APPROVED** [review]

**Pull request merged**